### PR TITLE
Fix missing data actions in 7.2-stable, 7.3-preview

### PR DESCRIPTION
**Pull request by daviddesberg**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

@maorleger  highlighted that 7.3-preview was not up to date with the latest changes (a DataAction enum) that @christothes had added some months ago. I also noticed that the custom role definition related data actions were missing from 7.2 stable, so I added those. 

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by heaths — COMMENTED** [review]
@Lusitanian could you use the https://aka.ms/openapihub diffing tools between 7.2 and 7.3-preview to make sure nothing else is missing, or was this the result of that? I suppose even diffing the text files in any text differ should suffice as well.

**Comment by Lusitanian** [comment]
@heaths - this was the result of a diff between 7.2 and 7.3-preview's rbac.json files - I'll check keys too

**Comment by Lusitanian** [comment]
@heaths looks like this closes the gap for RBAC, and keys doesn't appear to have any similar issues

**Review by heaths — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json [hunk 1]
```diff
@@ -524,7 +523,7 @@
       "required": [
         "properties"
       ],
-      "description": "Role definition creation parameters."
+      "description": "Role definition create parameters."
```
Comment (christothes) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json:526]: The old version seems to be more correct here - perhaps I am missing some context though.

**Review by christothes — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json (hunk 1 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json:526]: do you feel strongly? i can change it back - not sure that there was a reason for this, it's just how it read in 7.2 stable

Comment (christothes) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json:526]: not really - it just reads strangely to me.

**Comment by Lusitanian** [comment]
@heaths or @lmazuel  do you understand the validation that is failing re: breaking changes in the Go SDK here?

**Comment by heaths** [comment]
To ease, it merely reports this:

```
Diff with approved:
+	Package was removed
Diff with base:
+	Package was removed
```

I don't, no, but breaking change warnings are fairly common and this doesn't seem like a breaking change.

Given this is merely adding what was missing from the previous stable version, I see no reason to hold this up. @lmazuel can you merge this with your magic powers?

**Comment by Lusitanian** [comment]
ping @lmazuel @heaths any action for me here?

**Comment by heaths** [comment]
@jhendrixMSFT is there someone else available that can merge this? The breaking change - for KV - is expected here.

**Review by jlichwa — APPROVED** [review]

**Pull request merged**